api/src/test/java/com/epam/pipeline/manager/pipeline/RunScheduleManagerTest.java

```diff
@@ -103,214 +117,238 @@ public void setUp() {
+        doNothing().when(mockRunConfigurationScheduleProvider).verifySchedulable(anyLong());
+        doReturn(mockMessageHelper).when(mockRunConfigurationScheduleProvider).getMessageHelper();
+
         final PipelineRunScheduleVO runScheduleVO = getRunScheduleVO(RunScheduledAction.RESUME, CRON_EXPRESSION1);
+
         runScheduleManager.createSchedules(RUN_ID, ScheduleType.RUN_CONFIGURATION,
                 Collections.singletonList(runScheduleVO));
     }
 
     @Test
-    @WithMockUser(username = USER_OWNER)
     public void testUpdateRunSchedules() {
-        final List<RunSchedule> schedules =
-            runScheduleManager.createSchedules(RUN_ID, ScheduleType.PIPELINE_RUN,
-                    Arrays.asList(testRunScheduleVO, testRunScheduleVO2));
-
-        testRunScheduleVO.setScheduleId(schedules.get(0).getId());
+        testRunScheduleVO.setScheduleId(1L);
+        testRunScheduleVO2.setScheduleId(2L);
         testRunScheduleVO.setCronExpression(CRON_EXPRESSION2);
-        testRunScheduleVO2.setScheduleId(schedules.get(1).getId());
         testRunScheduleVO2.setCronExpression(CRON_EXPRESSION1);
 
-        final List<RunSchedule> updatedSchedules =
-            runScheduleManager.updateSchedules(RUN_ID, ScheduleType.PIPELINE_RUN,
-                    Arrays.asList(testRunScheduleVO, testRunScheduleVO2));
-        updatedSchedules.forEach(this::loadAndAssertSchedule);
-
-        final List<RunSchedule> loadRunSchedule = runScheduleManager
-                .loadAllSchedulesBySchedulableId(RUN_ID, ScheduleType.PIPELINE_RUN);
-        assertEquals(2, loadRunSchedule.size());
-        Mockito.verify(runScheduler, Mockito.times(2)).unscheduleRunSchedule(any());
-        // 2 for creation and 2 from update
-        Mockito.verify(runScheduler, Mockito.times(4)).scheduleRunSchedule(any());
+        doReturn(Arrays.asList(
+                createRunSchedule(1L, RUN_ID, RunScheduledAction.PAUSE, CRON_EXPRESSION1),
+                createRunSchedule(2L, RUN_ID, RunScheduledAction.RESUME, CRON_EXPRESSION2)))
+                .when(mockRunScheduleDao).loadAllRunSchedulesBySchedulableIdAndType(RUN_ID, ScheduleType.PIPELINE_RUN);
+
+        runScheduleManager.updateSchedules(RUN_ID, ScheduleType.PIPELINE_RUN,
+                Arrays.asList(testRunScheduleVO, testRunScheduleVO2));
+
+        verify(mockRunScheduler, times(2)).scheduleRunSchedule(any());
+        verify(mockRunScheduler, times(2)).unscheduleRunSchedule(any());
+        verify(mockRunScheduleDao, times(1)).updateRunSchedules(anyList());
     }
 
     @Test(expected = IllegalArgumentException.class)
-    @WithMockUser(username = USER_OWNER)
     public void testUpdateRunSchedulesFailsWithoutScheduleId() {
         runScheduleManager.createSchedules(RUN_ID, ScheduleType.PIPELINE_RUN,
-                        Collections.singletonList(testRunScheduleVO));
+                Collections.singletonList(testRunScheduleVO));
 
         testRunScheduleVO.setCronExpression(CRON_EXPRESSION2);
 
```

Comment:
Please do not try to create schedules before updating them. Please remember that all the dependent classes are mocks now so calling `createSchedules` basically does nothing.